packages/core-sdk/lib/Types/IConnector.ts

```diff
@@ -9,14 +9,21 @@ export interface IConnectorAdapters {
   multicall: IMulticallAdapter;
 }
 
+export interface OnNetworkChangeOptions {
+  tryReconnection: boolean;
+}
+
 export interface IConnector {
   isWallet: boolean;
   name: string;
   displayName: string;
   adapter: IConnectorAdapters | undefined;
   readonly config: IBlockchainConfig;
 
-  connect(): Promise<IConnectorAdapters>;
+  connect(
+    onNetworkChangeOptions?: OnNetworkChangeOptions
```

Comment:
I would rename it to options?: IConnectOptions as it is defining the connect options.